### DISCUSSION TITLE
Ways to enforce concurrency?

### DISCUSSION THREAD

**Discussion by ls-ivan-kiselev** [helmfile/helmfile#341] [Questions] [answered]
So, `--concurrencty` flag sets _"maximum number of concurrent downloads of release charts"_, but doesn't really enforce it.. So when I feed helmfile with a fairly big amount of charts, in my case ~100, what I observe - it still fetches them in batches of 2 (I think), but totally not 15 at time. resulting in quite some time to wait.

Is there a way to enforce concurrency?

The charts in my helmfile are not declared as dependencies of each other of whatnot, they shall be seen as independent pieces to be deployed.

The command I use is `helmfile -f helmfile.yaml lint --concurrency 15`, where all the helmcharts are defined under the same release in a single helmfile

**Comment by stek29** [comment]
from what I see it's actually hardcoded to 2:
https://github.com/helmfile/helmfile/blob/85ade797abf278cbb6acebe26c1b66b39c7a98ce/pkg/app/run.go#L71

**Comment by yxxhero** [comment, marked as answer]
fixed: https://github.com/helmfile/helmfile/pull/345
